src/util/duration.rs

```diff
@@ -102,6 +102,40 @@ pub fn parse_duration(ds: &str) -> Result<Duration, String> {
     }
 }
 
+/// display Duration in Prometheus format
+pub fn display_duration(duration: &Duration) -> String {
+    if duration.is_zero() {
+        return "0s".into();
+    }
+    let mut ms = duration.as_millis();
+    let mut ss = String::new();
+
+    let mut f = |unit: &str, mult: u128, exact: bool| {
+        if exact && ms % mult != 0 {
+            return;
+        }
+
+        let v = ms / mult;
+        if v > 0 {
+            ss.push_str(&format!("{v}{unit}"));
```

Comment:
I realize that we can just use `+=`
```suggestion
            ss += format!("{v}{unit}");
```